tests/test_e2e.py

```diff
@@ -1,34 +1,145 @@
+from os import path
+
 import pytest
+import re
 import subprocess
-from kubernetes import client, config, utils
+import yaml
+from kubernetes import client, config
+from create_from_yaml import create_from_yaml
 
+# TODO consider removing the raw stdout output of each test
 
-@pytest.mark.e2e
-def test_deny_all_traffic_to_an_application():
-    namespace = "01-deny-all"
-    config.load_kube_config()
-    k8s_client = client.ApiClient()
-    corev1 = client.CoreV1Api()
 
-    corev1.create_namespace(client.V1Namespace(metadata=client.V1ObjectMeta(name=namespace)))
-    utils.create_from_yaml(k8s_client,
-                           "e2e-manifests/01-deny-all-traffic-to-an-application.yml",
-                           namespace=namespace)
+def validate_illuminatio_was_successful(results_dict):
+    if results_dict["cases"] is None:
+        raise ValueError("Error: Invalid output, test cases seem to be missing")
+    cases_dict = results_dict["cases"]
+    for case in cases_dict:
+        raw_case = cases_dict[case]
+        for namespace in raw_case:
+            result = raw_case[namespace]
+            for ports in result:
+                success = result[ports]["success"]
+                if success is not True:
+                    raise ValueError("Error: One or more test cases of illuminatio have failed")
+
 
-    # ToDo add sleep or wait until all resources are up otherwise we have a race condition
-    # ToDo handle execptions
-    res = subprocess.run(["illuminatio", "run", "--runner-image=localhost:5000/illuminatio-runner:dev"],
+def create_resources_and_run_illuminatio(resource_manifest, results_yaml_file):
+    config.load_kube_config()
+    k8s_client = client.ApiClient()
+    create_from_yaml(k8s_client, resource_manifest, verbose=False, wait_until_ready=True)
+    # run illuminatio and store the results to a yaml file
+    res = subprocess.run(["illuminatio", "clean", "run", "--runner-image=localhost:5000/illuminatio-runner:dev",
+                          "-o", results_yaml_file],
                          capture_output=True,
-                         timeout=60)
+                         timeout=120)
+    assert res.returncode == 0
+
 
+def cleanup_resources(resource_manifest):
+    # delete illuminatio resources
+    res = subprocess.run(["illuminatio", "clean"], capture_output=True)
     assert res.returncode == 0
-    # ToDo evaluate result
     print(res.stdout)
 
-    # Clean up
-    res = subprocess.run(["illuminatio", "clean"], capture_output=True)
+    # delete test resources
+    res = subprocess.run(["kubectl", "delete", "-f", resource_manifest],
```

Comment:
You can leave this command if you just add the according labels to the manifests (then `illuminatio clean` will remove them for you)